static/src/javascripts/projects/commercial/modules/dfp/create-slots.ts

```diff
@@ -194,61 +193,103 @@ const createAdSlotElements = (
 		adSlot.setAttribute(attr, attrs[attr]);
 	});
 
-	adSlots.push(adSlot);
+	return adSlot;
+};
 
-	return adSlots;
+/**
+ * Split class names and prefix all with ad-slot--${className}
+ */
+const createClasses = (slotName: string, classes?: string): string[] =>
```

Comment:
Take or leave this suggestion but you could narrow the type here to something like (but it'll require an `as const` assertion in the function body):

```suggestion
const createClasses = (slotName: string, classes?: string): `ad-slot--${string}`[] =>
```

Reply:
You don’t need the `as const` assertion, you can simply:

1. Pass the expected return type to the `map` like `map<`ad-slot--${string}`>((className: string) => `ad-slot--${className}`)`
2. Then, use `Array<`ad-slot--${string}`>` as it’s a “complex type”

Reply:
That's much better @mxdvl